src/status_im/navigation/events.cljs

```diff
@@ -144,3 +145,9 @@
   {:events [:reload-status-nav-color]}
   [{:keys [db]} view-id]
   {:reload-status-nav-color-fx (or view-id (:view-id db))})
+
+(defn open-share
+  [_ [content]]
+  {:fx [[:open-share-fx content]]})
+
+(rf/reg-event-fx :open-share open-share)
```

Comment:
It's a bit unclear what `content` should/could be here without looking at it's usages or going to the rn-sharing docs. Maybe we can add a comment (ew) or destruct content as we pass it down the effect (less ew) ...

... or add a schema (which we shouldn't do just yet with events even though it's present in a few places already). Orr alternatively add a schema to the `rn/sharing` function.

CC: @J-Son89

Reply:
I think there's been some kind of pattern of passing a `{:url ...}` or `{:message ...}` as the content. That could be the initial definition of the schema, though I'll need look more into what types those values can be.

Reply:
For context, here's the API for using share in react-native: https://reactnative.dev/docs/share

Reply:
@seanstrom - if you look at later uses in the codebase it seems we are using "react-native-share" package.
If you look up uses of the ns react-native.share you'll find the other uses. So we should be consistent with out usages for sharing. There's also some places using this as an effect and others just directly call the function. 
We should really standardise the approach whatever the case 👍 

@clauxx I agree we should make some abstraction for this and create an api that suits are uses as it could be a lot simpler to use imo.
There's also a lot of functionality in the codebase which have this "Share to" feature as part of the native share screen, we need to add this to that as well - 
e.g https://github.com/status-im/status-mobile/issues/17324
 https://www.figma.com/file/q3kZPl7OpbB7btFK7HG0Mr/Sharing-for-Mobile?type=design&node-id=3315%3A317589&mode=design&t=ixP4lbjwwBc8ldob-1

Reply:
> It's a bit unclear what `content` should/could be here without looking at it's usages or going to the rn-sharing docs. Maybe we can add a comment (ew) or destruct content as we pass it down the effect (less ew) ...
> 
> ... or add a schema (which we shouldn't do just yet with events even though it's present in a few places already). Orr alternatively add a schema to the `rn/sharing` function.
> 
> CC: @J-Son89

on the topic of adding more schemas maybe we should open up a discussion next week if people are happy to add more schemas and figure out to what scope/ use cases? 🤔

Reply:
There's more that can be passed if you look [here](https://reactnative.dev/docs/share). Also seems like :url is only supported by iOS, so we should be careful with it.

Reply:
> on the topic of adding more schemas maybe we should open up a discussion next week if people are happy to add more schemas and figure out to what scope/ use cases? 🤔

That would be a good discussion topic to see what is the general vibe around expanding the use of schemas :+1:

Reply:
@seanstrom @J-Son89 We should definitely stick to one of them, and react-native-share seems to be offering more flexibility.

Reply:
It seems `react-native-share` provides a better cross-platform support, e.g. `:url` is okay for iOS and Android, so I'll update the code here to use our wrapper implementation of that package. Sound good? 

I'll also create a follow-up PR to update other usages of using Share to standardise on using dispatch, or I can do it in this PR, thoughts?

Reply:
either is fine for as long as it leaves for the share to functionality which I attached

Reply:
> either is fine for as long as it leaves for the share to functionality which I attached

Sorry I'm not sure I understand, are you asking that I mindful of that feature when changing the code (e.g. don't break / remove that functionality) or are you saying something else?

Reply:
Please look at the functionality in the issue/designs I attached above.

You'll see there is an option on the native share menu to share the data within status to a users share contacts.
I'm saying we should make this as a generic utility and whatever common wrapper you add here for the native share should consider we need to add the share-to feature later. Probably it's nothing complicated but it's at least worth being aware of.
If there's still confusion to what I mean here let's just discuss in messenger

Reply:
Ahh I see, I understand what you're saying 👍 
What I was planning on doing in this PR is updating the code to use the existing `react-native.share` namespace. That should keep things pointing to the same minimal bindings to `react-native-share`. So I wouldn't be creating a custom abstraction since it seems we can pass the data directly to share-sheet, and that way when [this issue](https://github.com/status-im/status-mobile/issues/17324) is finished we can attempt to extract some common code.